components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php

```diff
@@ -0,0 +1,417 @@
+                $settingKey = "logout_message_" . $lang_key;
+                if ($this->form->getInput($settingKey)) {
+                    $this->logoutSettings->set($settingKey, $this->form->getInput($settingKey));
+                }
+            }
+
+            if ($this->form->getInput('default_auth_mode')) {
+                $this->setting->set('default_auth_mode', $this->form->getInput('default_auth_mode'));
+            }
+
+            $this->tpl->setOnScreenMessage('success', $this->lng->txt("logout_information_settings_saved"), true);
+        }
+
+        $this->ctrl->redirect($this, 'show');
+    }
+
+    /**
+     * Init logout form
+     */
+    protected function initLogoutForm(): void
+    {
+        $this->form = new ilPropertyFormGUI();
+        $this->form->setFormAction($this->ctrl->getFormAction($this, 'saveLogoutInfo'));
+        $this->form->setTableWidth('80%');
+        $this->form->setTitle($this->lng->txt('logout_information'));
+
+        $this->form->addCommandButton('saveLogoutInfo', $this->lng->txt('save'));
+
+        if (!is_object($this->logoutSettings)) {
+            $this->logoutSettings = new ilSetting("logout_settings");
+        }
+
+        $logout_settings = $this->logoutSettings->getAll();
+        $languages = $this->lng->getInstalledLanguages();
+        $def_language = $this->lng->getDefaultLanguage();
+
+        foreach ($this->setDefLangFirst($def_language, $languages) as $lang_key) {
+            $add = "";
+            if ($lang_key === $def_language) {
+                $add = " (" . $this->lng->txt("default") . ")";
+            }
+
+            $textarea = new ilTextAreaInputGUI(
+                $this->lng->txt("meta_l_" . $lang_key) . $add,
+                'logout_message_' . $lang_key
+            );
+
+            $textarea->setRows(10);
+            $msg_logout_lang = "logout_message_" . $lang_key;
+
+            if (isset($logout_settings[$msg_logout_lang])) {
+                $textarea->setValue($logout_settings[$msg_logout_lang]);
+            }
+
+            $this->form->addItem($textarea);
+
+            unset($logout_settings["logout_message_" . $lang_key]);
+        }
+
+        foreach ($logout_settings as $key => $message) {
+            $lang_key = substr($key, strrpos($key, "_") + 1, strlen($key) - strrpos($key, "_"));
+
+            $textarea = new ilTextAreaInputGUI(
+                $this->lng->txt("meta_l_" . $lang_key),
+                'logout_message_' . $lang_key
+            );
+
+            $textarea->setRows(10);
+            $textarea->setValue($message);
+
+            if (!in_array($lang_key, $languages, true)) {
+                $textarea->setAlert($this->lng->txt("not_installed"));
+            }
+
+            $this->form->addItem($textarea);
+        }
+    }
+
+    /**
+     * returns an array of all installed languages, default language at the first position
+     * @param string $a_def_language Default language of the current installation
```

Comment:
```suggestion
     * @param string $a_def_language Default language of the current installation
```

Please define the shape of the arrays.

Reply:
done